File: dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/http-backend-akka-http/src/main/scala/controllers/HomeController.scala
/*
 * Copyright (C) Lightbend Inc. <https://www.lightbend.com>
 */

package controllers

import javax.inject._
import play.api._
import play.api.mvc._
@Singleton
class HomeController @Inject()(cc: ControllerComponents) extends AbstractController(cc) {

  def index() = Action { implicit request: Request[AnyContent] =>
    Ok("Successful response.")
  }
}

File: dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/jjwt-compatibility/app/controllers/HomeController.java
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package controllers;

import play.mvc.*;

public class HomeController extends Controller {

    public Result index(Http.Request req) {
        return ok("hello_world_" + req.session().get("foo").orElse("empty")) // This line tests session cookie parsing
            .addingToSession(req, "one", "two"); // This lines tests session cookie creation
    }

}
